ui/apps/dashboard/src/components/Insights/InsightsChat/input/InputField.tsx

```diff
@@ -0,0 +1,188 @@
+
+  const checkExpansion = useCallback(() => {
+    const textarea = textareaRef.current;
+    if (!textarea) return;
+    if (value.includes('\n')) {
+      if (!isExpanded) setIsExpanded(true);
+      return;
+    }
+    const tempSpan = document.createElement('span');
+    tempSpan.style.position = 'absolute';
+    tempSpan.style.visibility = 'hidden';
+    tempSpan.style.whiteSpace = 'nowrap';
+    if (typeof window !== 'undefined') {
+      const style = window.getComputedStyle(textarea);
+      tempSpan.style.fontSize = style.fontSize;
+      tempSpan.style.fontFamily = style.fontFamily;
+      tempSpan.style.fontWeight = style.fontWeight;
+      tempSpan.style.letterSpacing = style.letterSpacing;
+    }
+    tempSpan.textContent = value || textarea.placeholder;
+    document.body.appendChild(tempSpan);
+    const textWidth = tempSpan.getBoundingClientRect().width;
+    document.body.removeChild(tempSpan);
+    const availableWidth = textarea.getBoundingClientRect().width - 48;
+    const shouldExpand = textWidth >= availableWidth;
+    if (shouldExpand && !isExpanded) {
+      setIsExpanded(true);
+    } else if (!value.trim() && isExpanded) {
+      setIsExpanded(false);
+    }
+  }, [value, isExpanded]);
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      checkExpansion();
+    }, 0);
+    return () => clearTimeout(timeoutId);
+  }, [checkExpansion]);
+
+  useEffect(() => {
+    if (isExpanded && textareaRef.current) {
+      const textarea = textareaRef.current;
+      setTimeout(() => {
+        textarea.focus();
+        const length = textarea.value.length;
+        textarea.setSelectionRange(length, length);
+      }, 0);
+    }
+  }, [isExpanded]);
+
+  if (isExpanded) {
+    return (
+      <PromptInput onSubmit={onSubmit} className={className}>
+        <div className="flex flex-col">
+          <div className="relative mb-2 w-full">
+            <PromptInputTextarea
+              ref={textareaRef}
+              rows={5}
+              value={value}
+              onChange={onChange}
+              placeholder={placeholder}
+              disabled={disabled}
+              className="max-h-[30lh] min-h-[3lh] w-full resize-none px-4 py-0 pt-0 leading-6 placeholder:text-base"
+            />
+            <div className="from-surfaceBase pointer-events-none absolute left-0 right-0 top-0 h-3 bg-gradient-to-b to-transparent" />
+            {/* Bottom gradient overlay */}
+            <div className="from-surfaceBase pointer-events-none absolute bottom-0 left-0 right-0 h-3 bg-gradient-to-t to-transparent" />
+          </div>
+          <div className="flex items-center justify-end px-3 pb-3">
+            <div className="flex items-center gap-2">
+              <SendButton onClick={onSubmit} disabled={disabled || !value.trim()} />
+            </div>
+          </div>
+        </div>
+      </PromptInput>
+    );
+  }
+
+  return (
+    <PromptInput onSubmit={onSubmit} className={className}>
```

Comment:
As we previously spoke about, if we are making such a huge jump on height when adding a new line, this needs a transition or animation. Otherwise seems like something went wrong.
I still believe it would be simpler to just add one line to the bottom each time, up to a max height
https://github.com/user-attachments/assets/ac5db1d4-a6cc-4a96-ab1e-de6ef8837011
https://github.com/user-attachments/assets/d93a5291-6d51-4aa2-ad7c-f77bfabba1d7